arrow-schema/src/schema.rs

```diff
@@ -25,6 +25,32 @@ use crate::field::Field;
 use crate::{FieldRef, Fields};
 
 /// A builder to facilitate building a [`Schema`] from iteratively from [`FieldRef`]
+///
+/// # Example
+/// Create an entirely new Schema
+/// ```
+/// # use arrow_schema::*;
+/// let schema = Schema::builder()
+///   .with_field(Field::new("c1", DataType::Int64, false))
+///   .with_field(Field::new("c2", DataType::Utf8, false))
+///   .build();
+/// ```
+/// Create a new schema with a subset of fields from an existing schema
+/// ```
+/// # use arrow_schema::*;
+/// let schema = Schema::new(vec![
+///   Field::new("c1", DataType::Int64, false),
+///   Field::new("c2", DataType::Utf8, false),
+///  ]);
+///
+/// // Create a new schema with the same metdata, but only the second field
+/// let projected_schema = SchemaBuilder::from(&schema)
+///   .clear_fields()
+///   .with_field(schema.field(1).clone())
+///   .build();
```

Comment:
```suggestion
/// let projected_schema = SchemaBuilder::from(schema.metadata())
///   .with_field(schema.field(1).clone())
///   .build();
```

I wonder if we could make this work, it might be a bit cleaner?

Reply:
I agree -- will give it a try